<a href="https://colab.research.google.com/github/wellingtonsouza32/Python_MaterialEstudo/blob/master/exercicios_3.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

# Exercício 1.0

## Problema

In [2]:
from google.colab import drive
drive.mount('/content/drive')

Mounted at /content/drive


Considere um arquivo CSV com os seguintes dados de todas as copas do mundo até o ano de 2014. As informações do arquivo são:

* Year - Ano da copa
* Country - País sede
* Winner - Campeão
* Runners-Up - Vice-Campeão
* Third - Terceiro lugar
* Fourth - Quarto Lugar
* GoalsScored - Total de Gols
* QualifiedTeams - Quantidade de Times
* MatchesPlayed - Quantidade de Jogos
* Attendance - Público Total

Efetue a leitura desse arquivo, sem a utilização de módulos externos (Pandas), e crie um arquivo de saída com as seguintes informações (uma informação por linha):

* Soma de público das copas com anos final 0 (1930, 1950, etc)
* Quantidade total de gols entre as copas de 1954 e 1990, inclusive
* Média de público
* Média de gols por partida
* Quantidade de vezes em que o país sede foi campeão
* Quantidade de vezes em que o time do Brasil ficou entre uma das 4 primeiras posições
* Ano das edições em que o time da França finalizou em terceiro lugar
* Quantidade de vitórias por país, ordenada de forma crescente por número de títulos

## Análise

   Coloque a sua análise aqui

## Resolução e Resposta

In [3]:
with open('/content/drive/MyDrive/Colab Notebooks/world_cups.csv', "r") as textFile:
    lines = textFile.readlines()
    lines = list(map(lambda s: s.strip(), lines)) 
    print(lines)

['Year,Country,Winner,Runners-Up,Third,Fourth,GoalsScored,QualifiedTeams,MatchesPlayed,Attendance', '1930,Uruguay,Uruguay,Argentina,USA,Yugoslavia,70,13,18,590.549', '1934,Italy,Italy,Czechoslovakia,Germany,Austria,70,16,17,363.000', '1938,France,Italy,Hungary,Brazil,Sweden,84,15,18,375.700', '1950,Brazil,Uruguay,Brazil,Sweden,Spain,88,13,22,1.045.246', '1954,Switzerland,Germany FR,Hungary,Austria,Uruguay,140,16,26,768.607', '1958,Sweden,Brazil,Sweden,France,Germany FR,126,16,35,819.810', '1962,Chile,Brazil,Czechoslovakia,Chile,Yugoslavia,89,16,32,893.172', '1966,England,England,Germany FR,Portugal,Soviet Union,89,16,32,1.563.135', '1970,Mexico,Brazil,Italy,Germany FR,Uruguay,95,16,32,1.603.975', '1974,Germany,Germany FR,Netherlands,Poland,Brazil,97,16,38,1.865.753', '1978,Argentina,Argentina,Netherlands,Brazil,Italy,102,16,38,1.545.791', '1982,Spain,Italy,Germany FR,Poland,France,146,24,52,2.109.723', '1986,Mexico,Argentina,Germany FR,France,Belgium,132,24,52,2.394.031', '1990,Italy,G

1 - Soma de público das copas com anos final 0 (1930, 1950, etc)

In [20]:
import csv
filename = '/content/drive/MyDrive/Colab Notebooks/world_cups.csv'

soma = []

with open(filename, "r") as csv_file:
    reader = csv.reader(csv_file)
    next(reader)
    for linha in reader:
      if linha[0][3] == '0':
          string = linha[9].replace('.', '')
          soma.append(float(string))
    print (sum(soma))
       

 

8934841.0


2 - Quantidade total de gols entre as copas de 1954 e 1990

In [18]:
import csv
filename = '/content/drive/MyDrive/Colab Notebooks/world_cups.csv'

soma = []

with open(filename, "r") as csv_file:
    reader = csv.reader(csv_file)
    next(reader)
    for linha in reader:
      if ((linha[0] >= '1954') and (linha[0] <= '1990')):
          string = linha[6]
          soma.append(int(string))
    print (sum(soma))

1131


3 - Média de público:

In [34]:
import csv
filename = '/content/drive/MyDrive/Colab Notebooks/world_cups.csv'

soma = []
count = 0


with open(filename, "r") as csv_file:
    reader = csv.reader(csv_file)
    next(reader)
    for linha in reader:
      count += 1
      string = linha[9].replace('.', '')
      soma.append(float(string))
    print (sum(soma)/count)

1872882.35


4 - Média de gols por partida

In [37]:
import csv
filename = '/content/drive/MyDrive/Colab Notebooks/world_cups.csv'

soma = []
partida = []
count = 0


with open(filename, "r") as csv_file:
    reader = csv.reader(csv_file)
    next(reader)
    for linha in reader:
      count += 1
      string = linha[6].replace('.', '')
      soma.append(int(string))
      partida.append(int(linha[8]))
    print (sum(soma)/sum(partida))

2.8456937799043063


5 - Quantidade de vezes em que o país sede foi campeão

In [38]:
import csv
filename = '/content/drive/MyDrive/Colab Notebooks/world_cups.csv'


count = 0


with open(filename, "r") as csv_file:
    reader = csv.reader(csv_file)
    next(reader)
    for linha in reader:
      if linha[1] == linha[2]:
        print(f'País sede campeão {linha[1]}')
        count += 1
    print(f'Quantidade de vezes que o País sede foi Campeão {count}')
      

País sede campeão Uruguay
País sede campeão Italy
País sede campeão England
País sede campeão Argentina
País sede campeão France
Quantidade de vezes que o País sede foi Campeão 5


6-Quantidade de vezes em que o time do Brasil ficou entre uma das 4 primeiras posições:

In [52]:
import csv
filename = '/content/drive/MyDrive/Colab Notebooks/world_cups.csv'

count = 0

with open(filename, "r") as csv_file:
    reader = csv.reader(csv_file)
    next(reader)
    for linha in reader:
      if ((linha[2] == 'Brazil') or (linha[3] == 'Brazil') or (linha[4] == 'Brazil') or (linha[5] == 'Brazil')):
          count += 1
    print (f'Quantidade de vezes em que o time do Brasil ficou entre uma das 4 primeiras posições  {count}')

Quantidade de vezes em que o time do Brasil ficou entre uma das 4 primeiras posições  11


7 - Ano das edições em que o time da França finalizou em terceiro lugar:

In [56]:
import csv
filename = '/content/drive/MyDrive/Colab Notebooks/world_cups.csv'

count = 0
ano = []

with open(filename, "r") as csv_file:
    reader = csv.reader(csv_file)
    next(reader)
    for linha in reader:
      if linha[4] == 'France':
          count += 1
          ano.append(linha[0])
    print (f'Ano das edições em que o time da França finalizou em terceiro lugar {count} {ano}')

Ano das edições em que o time da França finalizou em terceiro lugar 2 ['1958', '1986']


8 - Quantidade de vitórias por país, classificada em ordem crescente do número de títulos

In [87]:
import csv
filename = '/content/drive/MyDrive/Colab Notebooks/world_cups.csv'

count = 0
campeao = []
dict = {}

with open(filename, "r") as csv_file:
    reader = csv.reader(csv_file)
    next(reader)
    for linha in reader:
      campeao.append(linha[2])
    for string in campeao:
      substring = string
      total = campeao.count(substring)
      dict[substring] = total
    for i in sorted(dict, key = dict.get):
      print(i,dict[i])


England 1
France 1
Spain 1
Germany 1
Uruguay 2
Argentina 2
Germany FR 3
Italy 4
Brazil 5


# Resposta

Soma de público das copas com final 0: 8934841   
Quantidade total de gols entre as copas de 1954 e 1990: 1131  
Média de público: 1872882.35  
Média de gols por partida: 2.85  
Quantidade de vezes em que o país sede foi campeão: 5  
Quantidade de vezes em que o time do Brasil ficou entre uma das 4 primeiras posições: 11  
Ano das edições em que o time da França finalizou em terceiro lugar: 1958, 1986  
Quantidade de vitórias por país, classificada em ordem crescente do número de títulos:  
* England:1
* France:1
* Spain:1
* Germany:1
* Uruguay:2
* Argentina:2
* Germany FR:3
* Italy:4
* Brazil:5 